src/cascadia/TerminalApp/TerminalWindow.cpp

```diff
@@ -0,0 +1,1262 @@
+// - The fully styled text run.
+static Documents::Run _BuildErrorRun(const winrt::hstring& text, const ResourceDictionary& resources)
+{
+    Documents::Run textRun;
+    textRun.Text(text);
+
+    // Color the text red (light theme) or yellow (dark theme) based on the system theme
+    auto key = winrt::box_value(L"ErrorTextBrush");
+    if (resources.HasKey(key))
+    {
+        auto g = resources.Lookup(key);
+        auto brush = g.try_as<winrt::Windows::UI::Xaml::Media::Brush>();
+        textRun.Foreground(brush);
+    }
+
+    return textRun;
+}
+
+namespace winrt::TerminalApp::implementation
+{
+    TerminalWindow::TerminalWindow(const TerminalApp::SettingsLoadEventArgs& settingsLoadedResult) :
+        _settings{ settingsLoadedResult.NewSettings() },
+        _initialLoadResult{ settingsLoadedResult }
+    {
+        // The TerminalPage has to be constructed during our construction, to
+        // make sure that there's a terminal page for callers of
+        // SetTitleBarContent
+        _root = winrt::make_self<TerminalPage>();
+        _root->WindowProperties(*this);
+        _dialog = ContentDialog{};
+
+        // For your own sanity, it's better to do setup outside the ctor.
+        // If you do any setup in the ctor that ends up throwing an exception,
+        // then it might look like App just failed to activate, which will
+        // cause you to chase down the rabbit hole of "why is App not
+        // registered?" when it definitely is.
+    }
+
+    // Method Description:
+    // - Implements the IInitializeWithWindow interface from shobjidl_core.
+    HRESULT TerminalWindow::Initialize(HWND hwnd)
+    {
+        // Pass commandline args into the TerminalPage. If we were supposed to
+        // load from a persisted layout, do that instead.
+        auto foundLayout = false;
+        if (const auto& layout = LoadPersistedLayout())
+        {
+            if (layout.TabLayout().Size() > 0)
+            {
+                std::vector<Settings::Model::ActionAndArgs> actions;
+                for (const auto& a : layout.TabLayout())
+                {
+                    actions.emplace_back(a);
+                }
+                _root->SetStartupActions(actions);
+                foundLayout = true;
+            }
+        }
+        if (!foundLayout)
+        {
+            _root->SetStartupActions(_appArgs.GetStartupActions());
+        }
+
+        // Check if we were started as a COM server for inbound connections of console sessions
+        // coming out of the operating system default application feature. If so,
+        // tell TerminalPage to start the listener as we have to make sure it has the chance
+        // to register a handler to hear about the requests first and is all ready to receive
+        // them before the COM server registers itself. Otherwise, the request might come
+        // in and be routed to an event with no handlers or a non-ready Page.
+        if (_appArgs.IsHandoffListener())

```

Comment:
same q as above, but more emphatically

Reply:
resolving as a part of the "commandline args are per-window" conversation